Change the way Oracle Java is installed and pull in upstream updates

jsf9k opened the pull request:
I also had to change the way Oracle Java is installed due to changes they made.  Oracle now requires you to create an account in order to download the Java archives that they distribute.  I created such an account and downloaded the archive to a private AWS S3 bucket so that installation can continue to be automated.

jsf9k committed:
Add changes from running pre-commit autoupdate

jsf9k committed:
Add changes from running pre-commit autoupdate

jsf9k committed:
Merge pull request #12 from cisagov/improvement/update_precommit

Add changes from running pre-commit autoupdate

jsf9k committed:
Remove the terraform-docs hook

terraform-docs is currently broken with respect to terraform 0.12.

jsf9k committed:
Remove the terraform-docs hook

terraform-docs is currently broken with respect to terraform 0.12.

jsf9k committed:
Merge pull request #13 from cisagov/bugfix/turn_off_terraform_docs_hook

Remove the terraform-docs hook

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/merge_in_upstream_changes

jsf9k committed:
Merge pull request #11 from cisagov/improvement/merge_in_upstream_changes

Merge in upstream changes from cisagov/skeleton-generic

jsf9k committed:
Configure prettier to ignore JSON files

JSON files are already being linted by the pretty-format-json hook.

jsf9k committed:
Configure prettier to ignore JSON files

JSON files are already being linted by the pretty-format-json hook.

jsf9k committed:
Merge pull request #14 from cisagov/improvement/tell_prettier_to_ignore_json_files

Configure prettier to ignore JSON files

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/pull_in_upstream_changes

jsf9k committed:
Merge pull request #12 from cisagov/improvement/pull_in_upstream_changes

Pull in upstream changes

jsf9k committed:
Add the docker extra to the installation of molecule

This gets rid of the need to install the docker python module
separately.

flake8 is now a dependency of the latest molecule, so there is no need
to install it separately either.

jsf9k committed:
Add the docker extra to the installation of molecule

This gets rid of the need to install the docker python module
separately.

flake8 is now a dependency of the latest molecule, so there is no need
to install it separately either.

jsf9k committed:
Merge pull request #13 from cisagov/improvement/better_installation_of_molecule

Add the docker extra to the installation of molecule

mcdonnnj committed:
Update two text blocks to use the console tag instead of the bash tag for formatting.

mcdonnnj committed:
Update two text blocks to use the console tag instead of the bash tag for formatting.

mcdonnnj committed:
Merge pull request #15 from cisagov/update_contributing_text_blocks

Update two text blocks to use the console tag instead of the bash tag…

mcdonnnj committed:
Merge upstream changes in skeleton-generic.

mcdonnnj committed:
Merge pull request #14 from cisagov/merge_skeleton-generic_updates

Merge skeleton generic updates

jsf9k committed:
Update pre-commit hooks via pre-commit autoupdate

jsf9k committed:
Update pre-commit hooks via pre-commit autoupdate

jsf9k committed:
Add config file for markdownlint

This config file turns off the "line too long" rule when within a code
block or a table.  Often there is no way to avoid going over 80 lines
in such cases.

jsf9k committed:
Add config file for markdownlint

This config file turns off the "line too long" rule when within a code
block or a table.  Often there is no way to avoid going over 80 lines
in such cases.

jsf9k committed:
Rename LICENSE.md to LICENSE, make the other changes that requires

The license text is not actually in Markdown, and adding the .md
extension actually makes it render a little differently than intended.

jsf9k committed:
Rename LICENSE.md to LICENSE, make the other changes that requires

The license text is not actually in Markdown, and adding the .md
extension actually makes it render a little differently than intended.

jsf9k committed:
Merge pull request #16 from cisagov/improvement/markdownlint_config

Add markdownlint config

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream_changes_from_skeleton_generic

jsf9k committed:
Merge pull request #15 from cisagov/improvement/upstream_changes_from_skeleton_generic

Merge in upstream changes from cisagov/skeleton-generic

jsf9k committed:
Add support for Debian Buster

Buster is scheduled to be released today.  Buster also includes the
freeipa-client package, while Stretch does not, so switching to Buster
would make our lives a lot easier going forward.

jsf9k committed:
Merge pull request #16 from cisagov/improvement/add_buster

Add support for Debian Buster

jsf9k committed:
Use a prerelease of molecule instead of the latest from master

The latest from master is currently broken, but the latest prerelease
has all the goodies we need.  Once a new release is made we will use
that instead.

jsf9k committed:
Use a prerelease of molecule instead of the latest from master

The latest from master is currently broken, but the latest prerelease
has all the goodies we need.  Once a new release is made we will use
that instead.

jsf9k committed:
Merge pull request #17 from cisagov/bugfix/upstream_molecule_broken

Use a prerelease of molecule instead of the latest from master

jsf9k committed:
Add updated Dockerfile.j2 and INSTALL.rst files

These skeleton files come from the latest release of molecule,
generated via the command "molecule init scenario".

jsf9k committed:
Add updated Dockerfile.j2 and INSTALL.rst files

These skeleton files come from the latest release of molecule,
generated via the command "molecule init scenario".

jsf9k committed:
Add latest .yamllint generated via molecule init scenario

The generated file contains the truthy: disable bit that was in our
old file.

jsf9k committed:
Add latest .yamllint generated via molecule init scenario

The generated file contains the truthy: disable bit that was in our
old file.

jsf9k committed:
Add fedora 29 and 30 to molecule configuration

jsf9k committed:
Trigger LGTM

jsf9k committed:
Merge pull request #18 from cisagov/improvement/updated_skeleton_files_from_latest_molecule

Add updated "molecule init scenario" files

jsf9k committed:
Add updates from running pre-commit autoupdate

jsf9k committed:
Add updates from running pre-commit autoupdate

jsf9k committed:
Merge pull request #17 from cisagov/improvement/updates_from_precommit_autoupdate

Add updates from running pre-commit autoupdate

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream_updates

jsf9k committed:
Merge pull request #19 from cisagov/improvement/upstream_updates

Upstream updates

jsf9k committed:
Change "company" name to the latest trend

The new name agrees with the naming guidance that was handed down by
Bobby Thompson.

jsf9k committed:
Change "company" name to the latest trend

The new name agrees with the naming guidance that was handed down by
Bobby Thompson.

jsf9k committed:
Merge pull request #20 from cisagov/change_company_name

Change "company" name to the latest trend

dav3r committed:
Add updates from running pre-commit autoupdate

dav3r committed:
Add updates from running pre-commit autoupdate

dav3r committed:
Merge pull request #18 from cisagov/improvement/updates_from_precommit_autoupdate

Add updates from running pre-commit autoupdate

dav3r committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/merge_in_upstream_changes

felddy committed:
Replace Travis-CI with GitHub actions.

felddy committed:
Replace Travis-CI with GitHub actions.

felddy committed:
Fix action shield link

felddy committed:
Disable ansible-lint until they fix their setup.py
See: https://github.com/ansible/ansible-lint/issues/590

felddy committed:
Disable ansible-lint until they fix their setup.py
See: https://github.com/ansible/ansible-lint/issues/590

dav3r committed:
Merge pull request #21 from cisagov/improvement/merge_in_upstream_changes

Merge in upstream changes

felddy committed:
Remove extra line.

felddy committed:
Remove extra line.

felddy committed:
Merge pull request #20 from cisagov/improvement/actions

Replace Travis-CI with GitHub actions

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/actions

felddy committed:
Add molecule run

felddy committed:
Add molecule run

felddy committed:
Remove white space that makes some of the team itchy.

felddy committed:
Remove white space that makes some of the team itchy.

felddy committed:
Merge pull request #22 from cisagov/improvement/actions

Replace Travis-CI with GitHub actions

mcdonnnj committed:
Re-enabled ansible-lint and ran pre-commit autoupgrade.

mcdonnnj committed:
Re-enabled ansible-lint and ran pre-commit autoupgrade.

mcdonnnj committed:
Merge pull request #21 from cisagov/pre-commit_autoupdate_2019-11-02

Re-enable ansible-lint and autoupgrade.

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/upstream

felddy committed:
Remove vestigial travis file.

felddy committed:
Remove vestigial travis file.

felddy committed:
Merge pull request #23 from cisagov/improvement/upstream

Pull in upstream: pre-commit updates, and clean out travis

jsf9k committed:
Change the way the Oracle-specific version of Java is installed

Oracle now requires you to create an account in order to download the
Java archives that they distribute.  I created such an account and
downloaded the archive here so that it can be copied over and
installed.

Also update the test code, since the two packages that are installed
have changed names slightly.  (The text "-local" has been appended to
their names.)  As a result the tests need to be altered slightly to
agree with this.

jsf9k committed:
Store the Oracle Java archive in a private AWS S3 bucket

The Oracle Java archive it too large for GitHub (without LFS), so it
will be stored in an AWS S3 bucket instead.

jsf9k committed:
Add boto3 as a testing requirement

This is necessary to connect to AWS S3 and retrieve the Oracle Java
archive.

jsf9k committed:
Merge remote-tracking branch 'skeleton-ansible-role/develop' into improvement/upstream-updates

jsf9k commented:
I has to do a complicated rebase in order to consolidate some commits, since GitHub wouldn't let me push as long as the large Oracle Java archive file was in the history.  In the process I somehow goofed up the commit history of this repo a little bit.

jsf9k committed:
Add secret values for AWS creds as environment variables

jsf9k commented:
OK @cisagov/team-ois, this one is ready for review now.

dav3r commented on tasks/main.yml:
```diff
@@ -25,8 +31,24 @@
 
+- name: Copy the Oracle Java archive from our S3 bucket
+  aws_s3:
+    bucket: oracle-java-archives
```
Since bucket names must be unique across AWS, shouldn't we make this bucket name a variable, to make things a bit easier for future users of this role (outside of our organization)?

jsf9k replied:
Yes.  I will do that now.

jsf9k replied:
Remedied in 935a1b9.

dav3r replied:
Looks great!  🔭

dav3r reviewed (approved):
Nice work @jsf9k - I have but one minor comment for you to consider.
☕️ 🎢

jsf9k committed:
Add Terraform code to create GitHub Actions user

The GitHub Actions user needs to be able to pull the Oracle Java
archive from our private S3 bucket.

jsf9k committed:
Make the name of the bucket where the download is stored a variable

felddy reviewed (approved):
Strong work!  Your javascript needs some polishing though.   ;)
